bson/primitive/decimal.go

```diff
@@ -52,13 +59,13 @@ func (d Decimal128) String() string {
 	if d.h>>61&3 == 3 {
 		// Bits: 1*sign 2*ignored 14*exponent 111*significand.
 		// Implicit 0b100 prefix in significand.
-		e = int(d.h>>47&(1<<14-1)) - 6176
+		e = int(d.h>>47&(1<<14-1)) + MinDecimal128Exp
```

Comment:
I think this can be done:

Probably best to extract out the `+ MinDecimal128Exp` from the cases here and instead, after the case, add it only a single time:
```go
if ... {
    ...
} else {
    ...
}
e += MinDecimal128Exp
```